types/spreedly-iframe-browser/index.d.ts

```diff
@@ -0,0 +1,308 @@
+        metadata: { [key: string]: string };
+        callback_url: string | null;
+        last_four_digits: string | null;
+        first_six_digits: string | null;
+        card_type: SpreedlyCardType;
+        first_name: string | null;
+        last_name: string | null;
+        month: number | null;
+        year: number | null;
+        address1: string | null;
+        address2: string | null;
+        city: string | null;
+        state: string | null;
+        zip: string | null;
+        country: string | null;
+        phone_number: string | null;
+        company: string | null;
+        full_name: string | null;
+        eligible_for_card_updater: boolean;
+        shipping_address1: string | null;
+        shipping_address2: string | null;
+        shipping_city: string | null;
+        shipping_state: string | null;
+        shipping_zip: string | null;
+        shipping_country: string | null;
+        shipping_phone_number: string | null;
+        payment_method_type: 'credit_card' | 'bank_account' | 'apple_pay' | 'google_pay' | 'third_party_token';
+        errors: any[];
+        fingerprint: string | null;
+        verification_value: string | null;
+        number: string | null;
+    }
+}
+
+/**
+ * Create a new, independent, instance of the iFrame. It will be created alongside the default instance, already exposed as Spreedly.
+ * @see https://docs.spreedly.com/reference/iframe/v1/#spreedlypaymentframe
+ */
+declare class SpreedlyPaymentFrame {
+    /**
+     * Initialize the iFrame library. This must be the first call made to iFrame and will trigger the loading of the iFrame fields.
+     * Triggers `ready` when the iFrame is ready for more configurartion.
+     * @see https://docs.spreedly.com/reference/iframe/v1/#init
+     *
+     * @param environmentKey - The key of the Spreedly environment where the payment method should be tokenized.
+     * @param options - Map of initialization options. HTML `id`s where iFrames should be rendered.
+     */
+    init(environmentKey: string, options: spreedly.InitOptions): void;
+
+    /**
+     * Reload the iFrame library. This resets and re-initializes all iFrame elements and state.
+     * @see https://docs.spreedly.com/reference/iframe/v1/#reload
+     */
+    reload(): void;
+
+    /**
+     * Remove all event handlers currently registered via the on function.
+     * @see https://docs.spreedly.com/reference/iframe/v1/#removehandlers
+     */
+    removeHandlers(): void;
+
+    /**
+     * Trigger tokenization of the entered credit card.
+     * On successful tokenization, the `paymentMethod` event will be triggered. On failure, the `errors` event will be triggered.
+     * @see https://docs.spreedly.com/reference/iframe/v1/#tokenizecreditcard
+     *
+     * @param additionalFields - Map of additional payment method fields to store alongside tokenized card.
+     */
+    tokenizeCreditCard(additionalFields?: spreedly.TokenizeCreditCardAdditionalFields): void;
```

Comment:
is this optional really?
https://docs.spreedly.com/reference/iframe/v1/#tokenizecreditcard
also the options state `required` for some props

Reply:
Indeed, you are right! I have made it required. As for the required properties, it is required to either pass `full_name` or `first_name` and `last_name`. I have pushed the needed changes for that too.